### PR TITLE
KB: Add filesystem scanning and repairing HOWTO

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
related to https://github.com/harvester/harvester/issues/3258

### CODE REVIEW

**Comment by Martin-Weiss** [comment]
As a general feedback - I guess we should also explain the process of booting the VM "temporarily" with an OS Repair ISO as I guess we might be missing proper NTFS repair tools in a container image..

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+![finding the target VM](./imgs/finding_the_target_vm.png)
+
+- Stop the target VM
+
+![Stop the target VM](./imgs/stop_the_target_vm.png)
+
+Now, the VM stopped, and related volumes are detached. Then we go to the longhorn UI to backup this volume.
+
+- Find the related volume of VM. Then, attach it again for scanning.
```
Comment (bk201) [inline kb/2023-02-01/scan_and_repair_filesystem.md:30]: Need a precise way to find the volume. One way is (for v1.1.1):
- Enable "Developer Tools & Features" (like https://docs.harvesterhci.io/v1.1/troubleshooting/harvester#access-embedded-rancher-and-longhorn-dashboards)
- Click VM for detail.
- Click the Volumes tab. There is an arrow icon called "check volume details", that should lead to LH volume page.

**Review by bk201 — APPROVED** [review]
LGTM!

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+authors:
+  - name: Vicente Cheng
+    title: Senior Software Engineer
+    url: https://github.com/Vicente-Cheng
+    image_url: https://github.com/Vicente-Cheng.png
+tags: [storage, longhorn, root, filesystem]
+hide_table_of_contents: false
+---
+
+In the early version of harvester (v1.0.3 and before versions), the longhorn volume sometimes corrupts when rebuilt (reference: [Analysis: Potential Data/Filesystem Corruption](https://longhorn.io/kb/troubleshooting-volume-filesystem-corruption/#solution)). In the harvester v1.1.0 and later versions, the longhorn team already fix this problem. Therefore, you could scan your filesystem of the VM by following the steps and repairing it (if needed).
```
Comment (guangbochen) [inline kb/2023-02-01/scan_and_repair_filesystem.md:14]:
```suggestion
In earlier versions of Harvester (v1.0.3 and prior), Longhorn volumes sometimes get corrupted when rebuilding (reference: [Analysis: Potential Data/Filesystem Corruption](https://longhorn.io/kb/troubleshooting-volume-filesystem-corruption/#solution)). In Harvester v1.1.0 and later versions, the Longhorn team has fixed this problem. So, you can follow the steps below to scan the VM's filesystem and repair it if needed.

```

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+## Stop The VM And Backup Volume
+
+Before we scan the filesystem, we recommend you back up your volume. For example, refer to the following steps to stop the VM and backup the volumes.
```
Comment (guangbochen) [inline kb/2023-02-01/scan_and_repair_filesystem.md:18]:
```suggestion
Before we scan the filesystem, we recommend you back up the volume first. For example, refer to the following steps to stop the VM and backup the volume.
```

**Comment by guangbochen** [comment]
And loop @btat for some proofreading, thanks.

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+- Find the target VM
+
+![finding the target VM](./imgs/finding_the_target_vm.png)
+
+- Stop the target VM
+
+![Stop the target VM](./imgs/stop_the_target_vm.png)
+
+Now, the VM stopped, and related volumes are detached. Then we go to the longhorn UI to backup this volume.
```
[the remark this answers is not in the document]
Comment (Vicente-Cheng) [reply kb/2023-02-01/scan_and_repair_filesystem.md:28]: Yes, we could do that. But we still need to go to the LH UI to attach and detach manually. I thought that would be more consistent for combining the operations like snapshot -> attach -> scan/repair -> detach on LH UI. WDUT?

[the remark this answers is not in the document]
Comment (LucasSaintarbor) [reply kb/2023-02-01/scan_and_repair_filesystem.md:29]: test

**Comment by btat** [comment]
> And loop @btat for some proofreading, thanks.

Hi @guangbochen, @LucasSaintarbor from my name will help take a look.

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+---
+
+In earlier versions of Harvester (v1.0.3 and prior), Longhorn volumes sometimes get corrupted when rebuilding (reference: [Analysis: Potential Data/Filesystem Corruption](https://longhorn.io/kb/troubleshooting-volume-filesystem-corruption/#solution)). In Harvester v1.1.0 and later versions, the Longhorn team has fixed this problem. So, you can follow the steps below to scan the VM's filesystem and repair it if needed.
+
+
+## Stop The VM And Backup Volume
+
+Before we scan the filesystem, we recommend you back up the volume first. For example, refer to the following steps to stop the VM and backup the volume.
+
+- Find the target VM
```
Comment (LucasSaintarbor) [inline kb/2023-02-01/scan_and_repair_filesystem.md:21]:
```suggestion
- Find the target VM.
```

Comment (LucasSaintarbor) [inline kb/2023-02-01/scan_and_repair_filesystem.md:21]:
```suggestion
- Find the target VM.
```

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+![finding the target VM](./imgs/finding_the_target_vm.png)
+
+- Stop the target VM
```
Comment (LucasSaintarbor) [inline kb/2023-02-01/scan_and_repair_filesystem.md:25]:
```suggestion
- Stop the target VM.
```

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+![Stop the target VM](./imgs/stop_the_target_vm.png)
+
+Now, the VM stopped, and related volumes are detached. Then we go to the longhorn UI to backup this volume.
+
+- Enable `Developer Tools & Features` (Preferences -> Enable Developer Tools & Features)
```
Comment (LucasSaintarbor) [inline kb/2023-02-01/scan_and_repair_filesystem.md:31]:
```suggestion
- Enable `Developer Tools & Features` (Preferences -> Enable Developer Tools & Features).
```

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+
+![Preferences then enable developer mode](./imgs/preferences_enable_developer_mode.png)
+![Enable the developer mode](./imgs/enable_the_developer_mode.png)
+
+- Go to edit config page of VM
```
Comment (LucasSaintarbor) [inline kb/2023-02-01/scan_and_repair_filesystem.md:36]:
```suggestion
- Go to edit config page of VM.
```

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+tags: [storage, longhorn, root, filesystem]
+hide_table_of_contents: false
+---
+
+In earlier versions of Harvester (v1.0.3 and prior), Longhorn volumes sometimes get corrupted when rebuilding (reference: [Analysis: Potential Data/Filesystem Corruption](https://longhorn.io/kb/troubleshooting-volume-filesystem-corruption/#solution)). In Harvester v1.1.0 and later versions, the Longhorn team has fixed this problem. So, you can follow the steps below to scan the VM's filesystem and repair it if needed.
+
+
+## Stop The VM And Backup Volume
+
+Before we scan the filesystem, we recommend you back up the volume first. For example, refer to the following steps to stop the VM and backup the volume.
```
Comment (LucasSaintarbor) [inline kb/2023-02-01/scan_and_repair_filesystem.md:19]:
```suggestion
Before you scan the filesystem, it is recommend you back up the volume first. For an example, refer to the following steps to stop the VM and backup the volume.
```

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+
+- Find the target VM
+
+![finding the target VM](./imgs/finding_the_target_vm.png)
+
+- Stop the target VM
+
+![Stop the target VM](./imgs/stop_the_target_vm.png)
+
+Now, the VM stopped, and related volumes are detached. Then we go to the longhorn UI to backup this volume.
```
Comment (LucasSaintarbor) [inline kb/2023-02-01/scan_and_repair_filesystem.md:29]:
```suggestion
The target VM is now stopped and the related volumes are detached. Now go to the Longhorn UI to backup this volume.
```

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+authors:
+  - name: Vicente Cheng
+    title: Senior Software Engineer
+    url: https://github.com/Vicente-Cheng
+    image_url: https://github.com/Vicente-Cheng.png
+tags: [storage, longhorn, root, filesystem]
+hide_table_of_contents: false
+---
+
+In earlier versions of Harvester (v1.0.3 and prior), Longhorn volumes sometimes get corrupted when rebuilding (reference: [Analysis: Potential Data/Filesystem Corruption](https://longhorn.io/kb/troubleshooting-volume-filesystem-corruption/#solution)). In Harvester v1.1.0 and later versions, the Longhorn team has fixed this problem. So, you can follow the steps below to scan the VM's filesystem and repair it if needed.
```
Comment (LucasSaintarbor) [inline kb/2023-02-01/scan_and_repair_filesystem.md:14]:
```suggestion
In earlier versions of Harvester (v1.0.3 and prior), Longhorn volumes sometimes get corrupted when rebuilding (reference: [Analysis: Potential Data/Filesystem Corruption](https://longhorn.io/kb/troubleshooting-volume-filesystem-corruption/#solution)). In Harvester v1.1.0 and later versions, the Longhorn team has fixed this problem. This article covers the steps you can take to scan the VM's filesystem and repair it if needed.
```

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+
+- Choose the attached node. 
+
+![choose the attached node](./imgs/choose_the_attached_node.png)
+
+- Check the volume attached and take snapshot on this volume page.
+
+![take snapshot on volume page](./imgs/take_snapshot_on_volume_page.png)
+
+- Check the snapshot is ready.
```
Comment (LucasSaintarbor) [inline kb/2023-02-01/scan_and_repair_filesystem.md:56]:
```suggestion
- Confirm that the snapshot is ready.
```

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+- Enable `Developer Tools & Features` (Preferences -> Enable Developer Tools & Features)
+
+![Preferences then enable developer mode](./imgs/preferences_enable_developer_mode.png)
+![Enable the developer mode](./imgs/enable_the_developer_mode.png)
+
+- Go to edit config page of VM
+
+![goto edit config page of VM](./imgs/goto_vm_edit_config_page.png)
+
+- Choose volume and Click `Check volume details`
```
Comment (LucasSaintarbor) [inline kb/2023-02-01/scan_and_repair_filesystem.md:40]:
```suggestion
- Go to **Volumes** and select `Check volume details.`
```

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+
+![link to longhorn volume page](./imgs/link_to_longhorn_volume.png)
+
+- Attach this volume again.
```
Comment (LucasSaintarbor) [inline kb/2023-02-01/scan_and_repair_filesystem.md:44]: Is there a name for this dropdown menu on the "Check volume details" page? 
Screen Shot 2023-02-10 at 3 28 40 PM

Comment (LucasSaintarbor) [inline kb/2023-02-01/scan_and_repair_filesystem.md:44]: Go to `?` and select `Attach` to attach the volume again.

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+
+![attach this volume again](./imgs/attach_this_volume_again.png)
+
+- Choose the attached node.
```
Comment (LucasSaintarbor) [inline kb/2023-02-01/scan_and_repair_filesystem.md:48]:
```suggestion
- Select the attached node. 
```

File: kb/2023-02-01/scan_and_repair_filesystem.md (hunk 12 above)
Comment (LucasSaintarbor) [inline kb/2023-02-01/scan_and_repair_filesystem.md:56]:
```suggestion
- Confirm the snapshot is ready.
```

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+
+- Check the volume attached and take snapshot on this volume page.
+
+![take snapshot on volume page](./imgs/take_snapshot_on_volume_page.png)
+
+- Check the snapshot is ready.
+
+![check the snapshot is ready](./imgs/check_the_snapshot_is_ready.png)
+
+Now we complete the backup volume. Next, we scan and repair the root filesystem.
```
Comment (LucasSaintarbor) [inline kb/2023-02-01/scan_and_repair_filesystem.md:60]:
```suggestion
Now that you completed the volume backup, you need to scan and repair the root filesystem.
```

File: kb/2023-02-01/scan_and_repair_filesystem.md (hunk 11 above)
Comment (Vicente-Cheng) [reply kb/2023-02-01/scan_and_repair_filesystem.md:14]: Hi @LucasSaintarbor, the first half of the sentences changed the original meaning.
I would like to present that the old version longhorn sometime makes replicas corrupted when rebuilding.
How would we improve the sentences here?

File: kb/2023-02-01/scan_and_repair_filesystem.md (hunk 14 above)
Comment (Vicente-Cheng) [reply kb/2023-02-01/scan_and_repair_filesystem.md:44]: There is no name for this dropdown menu, only an icon here.
This page is the page that when we click the `Check volume details` would link to.

Comment (Vicente-Cheng) [reply kb/2023-02-01/scan_and_repair_filesystem.md:44]: Hmm, only a dropdown menu icon here. How about
`Click the dropdown menu on the right side and attach this volume again`?

File: kb/2023-02-01/scan_and_repair_filesystem.md (hunk 11 above)
Comment (LucasSaintarbor) [reply kb/2023-02-01/scan_and_repair_filesystem.md:14]: Hi @Vicente-Cheng, thanks for the feedback, is the first sentence correct? Does it summarize the issue?

**Review by LucasSaintarbor — CHANGES_REQUESTED** [review]
Hi @Vicente-Cheng may you review these suggestions? Please let me know if any of the changes affect the meaning

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 17]
```diff
@@ -0,0 +1,125 @@
...
+
+- Click the `⋮` button and select **Edit Config** to edit the config page of the VM.
+
+![goto edit config page of VM](./imgs/goto_vm_edit_config_page.png)
+
+- Go to the `Volumes` tab and select `Check volume details.`
+
+![link to longhorn volume page](./imgs/link_to_longhorn_volume.png)
+
+- Attach this volume again.
```
Comment (LucasSaintarbor) [inline kb/2023-02-01/scan_and_repair_filesystem.md:44]:
```suggestion
- Click the dropdown menu on the right side and select 'Attach' to attach the volume again. 
```

File: kb/2023-02-01/scan_and_repair_filesystem.md [hunk 18]
```diff
@@ -0,0 +1,125 @@
...
+
+- Detach volume on longhorn UI.
+
+![detach volume on longhorn UI](./imgs/detach_volume.png)
+
+- Start VM again.
+
+![Start VM again](./imgs/start_vm_again.png)
+
+Now, your VM should work normally.
```
Comment (LucasSaintarbor) [inline kb/2023-02-01/scan_and_repair_filesystem.md:125]:
```suggestion
Your VM should now work normally.
```

**Review by LucasSaintarbor — CHANGES_REQUESTED** [review]
Hi @Vicente-Cheng, I made a two more small revisions. Looks good to me.

File: kb/2023-02-01/scan_and_repair_filesystem.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Vicente-Cheng) [reply kb/2023-02-01/scan_and_repair_filesystem.md]: @guangbochen, friendly ping. WDUT for this?

File: kb/2023-02-01/scan_and_repair_filesystem.md (hunk 14 above)
Comment (Vicente-Cheng) [reply kb/2023-02-01/scan_and_repair_filesystem.md]: hi @LucasSaintarbor, could you recheck this part?  Thanks!

Comment (Vicente-Cheng) [reply kb/2023-02-01/scan_and_repair_filesystem.md]: @LucasSaintarbor, ditto. Thanks for giving a review!

**Review by guangbochen — APPROVED** [review]
LGTM, and please help squash to merge.

File: kb/2023-02-01/scan_and_repair_filesystem.md (hunk 14 above)
Comment (LucasSaintarbor) [reply kb/2023-02-01/scan_and_repair_filesystem.md]: @Vicente-Cheng confirming if this is step good? Thanks

Comment (Vicente-Cheng) [reply kb/2023-02-01/scan_and_repair_filesystem.md]: hi @LucasSaintarbor.
Ah, I see it. We have already fixed it. Thanks!!

**Pull request merged**